### PR TITLE
BUG: GH11517 add multiindex column names after describe()

### PR DESCRIPTION
**Pull request by varunentropy**
Hi,

This fixes the bug mentioned in https://github.com/pydata/pandas/issues/11517
I have added column names after the describe operation. 
Request you to review the change and merge it. 

Thanks

### CODE REVIEW

**Comment by hayd** [comment]
Please add a test for this :)

**Comment by varun-kr** [comment]
Hi,

I have added test case for that . 
Request you to review the change .

File: pandas/tests/test_generic.py [hunk 1]
```diff
@@ -1471,6 +1471,16 @@ def test_describe_typefiltering_groupby(self):
         self.assertTrue(G.describe(include=['number', 'object']).shape == (22, 3))
         self.assertTrue(G.describe(include='all').shape == (26, 4))
 
+    # GH 11517
+    def test_describe_multi_index_df_column_names(self):
+        df = pd.DataFrame({'A': ['foo', 'bar', 'foo', 'bar', 'foo', 'bar', 'foo', 'foo'],
```
Comment (jreback) [inline pandas/tests/test_generic.py]: do the same tests, but just group on A (to simulate a non-hierarchical index).

File: pandas/core/generic.py [hunk 2]
```diff
@@ -4516,6 +4516,9 @@ def describe_1d(data, percentiles):
                 if name not in names:
                     names.append(name)
         d = pd.concat(ldesc, join_axes=pd.Index([names]), axis=1)
+        # if hierarchical index has column names, add column names to d.
+        if self.columns.names:
+            d.columns.names = self.columns.names
```
Comment (jreback) [inline pandas/core/generic.py]: This should always be done (no need for the if)

further, should be ``d.columns.names = data.columns.names``

**Comment by jreback** [comment]
pls add a whatsnew note for 0.17.1

**Comment by varun-kr** [comment]
Thanks for comments.
I have modified test case and also added comment to whatsnew 0.17.1
Please review..

File: pandas/tests/test_generic.py [hunk 3]
```diff
@@ -1471,6 +1471,22 @@ def test_describe_typefiltering_groupby(self):
...
+                           'B': ['one', 'one', 'two', 'three', 'two', 'two', 'one', 'three'],
+                           'C': np.random.randn(8),
+                           'D': np.random.randn(8)})
+        # test for hierarchical index
```
Comment (jreback) [inline pandas/tests/test_generic.py]: great ,use a blank line between different cases (e.g. before the comment).

move the isssue number inside the test, and provide a description of what this tests is doing

**Comment by jreback** [comment]
couple of comments. pls squash when done. ping on green.

**Comment by varun-kr** [comment]
I have made the changes. 
Request you to please review and merge.

**Comment by jreback** [comment]
looks good. pls squash to a single commit. ping when green.

**Comment by varun-kr** [comment]
Done. Please check.

**Comment by jreback** [comment]
pls squash: http://pandas-docs.github.io/pandas-docs-travis/contributing.html#combining-commits

**Comment by varun-kr** [comment]
Apologies for not getting it right the first time
Please check.

**Comment by jreback** [comment]
thank you sir!

**Comment by dragoljub** [comment]
@varun-kr @jreback 

Thanks for fixing this bug. :+1:  I'll have to get on the bandwagon and submit some pull requests also.

**Pull request merged**